src/app/child-dev-project/attendance/attendance.service.spec.ts

```diff
@@ -60,4 +63,45 @@ describe("AttendanceService", () => {
       )
     ).toBeTrue();
   });
+
+  it("getAllActivityAttendancesForPeriod", async () => {
```

Comment:
```suggestion
  it("should return all attendances for an activity", async () => {
```

Reply:
This test does not show, that the attendances are filtered by date

Reply:
Thanks for pushing me regarding unit tests. Added basic test cases.